### ISSUE TITLE
ToFutureValue stopped working in 2.1.0

### ISSUE THREAD

**Issue by starkpl** [maca88/NHibernate.Extensions#18]
Hi. Recently I updated to NHibernate.Extensions 2.1.0 to benefit from .NET Standard support in my .NET Core project. Everything works except the `ToFutureValue` method.
This is the (simplified) usage of the method:
```
var query = Session.Query<MyEntity>()
                .Include(s => s.Relation1)
                .Include(x => x.Relation2)
                .Where(somePredicate);
var futureCount =  query.ToFutureValue(q => q.LongCount());
```
It worked with NHibernate.Extensions 2.0.0, but now with 2.1.0 it throws:
```
System.ArgumentException : Expression of type 'System.Int64' cannot be used for parameter of type 'System.Linq.IQueryable`1[MyEntity]' of method 'NHibernate.Linq.INhFetchRequest`2[Relation2] Fetch[MyEntity,Relation2](System.Linq.IQueryable`1[MyEntity], System.Linq.Expressions.Expression`1[System.Func`2[Relation2]])'
Parameter name: arg0
   at System.Dynamic.Utils.ExpressionUtils.ValidateOneArgument(MethodBase method, ExpressionType nodeKind, Expression arguments, ParameterInfo pi, String methodParamName, String argumentParamName, Int32 index)
   at System.Linq.Expressions.Expression.Call(MethodInfo method, Expression arg0, Expression arg1)
   at NHibernate.Extensions.Linq.IncludeQueryProvider.FetchFromPath(ExpressionInfo expressionInfo, String path)
   at System.Linq.Enumerable.Aggregate[TSource,TAccumulate](IEnumerable`1 source, TAccumulate seed, Func`3 func)
   at NHibernate.Extensions.Linq.IncludeQueryProvider.GetExpressions(Expression queryExpression)
   at NHibernate.Extensions.Linq.IncludeQueryProvider.ExecuteQueryTreeFutureValue[T](Expression queryExpression)
```
I am using NHibernate 5.1.3.
Any idea what causes the exception?

**Comment by maca88** [comment]
Thanks for reporting the issue. The issue is now fixed and a new version `2.1.1` was released, please try with the new version if it works for you.

**Comment by starkpl** [comment]
It works with the new version, thanks a lot!
